### DISCUSSION TITLE
OpenAPI UI Scaffolding

### DISCUSSION THREAD

**Discussion by amalsharief** [oqtane/oqtane.framework#1979] [Ideas]
Hi,
Is there any future planning for generating oqtane components and pages from OpenAPI specifications?

thank yo in advance,

amalsharief

**Comment by hishamco** [comment]
Could you please elaborate?

**Reply by amalsharief** [reply]
Having a REST Web API (swagger) used for CRUD operations; a Blazor page (component ) for data entry could be generated automatically.

**Reply by hishamco** [reply]
We already support Swagger, but I didn't understand how the send part "a Blazor page (component ) for data entry could be generated" are related to Open API?

**Comment by sbwalker** [comment]
@amalsharief at this point there is no plan for the core framework to offer this service - however it sounds like a good opportunity for a third party module.
